test/integration/acceptance/create_asset_test.cpp

```diff
@@ -0,0 +1,258 @@
+#include "framework/specified_visitor.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+class CreateAssetFixture : public AcceptanceFixture {
+ public:
+  auto makeUserWithPerms(const interface::RolePermissionSet &perms = {
+                             interface::permissions::Role::kCreateAsset}) {
+    return AcceptanceFixture::makeUserWithPerms(perms);
+  }
+
+  const interface::types::AssetNameType kAssetName = "newcoin";
+  const interface::types::PrecisionType kPrecision = 1;
+  const interface::types::PrecisionType kNonDefaultPrecision = kPrecision + 17;
+  const interface::types::DomainIdType kNonExistingDomain = "nonexisting";
+  const std::vector<interface::types::AssetNameType> kIllegalAssetNames = {
+      "",
+      " ",
+      "   ",
+      "A",
+      "assetV",
+      "asSet",
+      "asset%",
+      "^123",
+      "verylongassetname_thenameislonger",
+      "verylongassetname_thenameislongerthanitshouldbe",
+      "assset-01"};
+
+  const std::vector<interface::types::DomainIdType> kIllegalDomainNames = {
+      "",
+      " ",
+      "   ",
+      "9start.with.digit",
+      "-startWithDash",
+      "@.is.not.allowed",
+      "no space is allowed",
+      "endWith-",
+      "label.endedWith-.is.not.allowed",
+      "aLabelMustNotExceeds63charactersALabelMustNotExceeds63characters",
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPad."
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPad."
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPad."
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPadP",
+      "257.257.257.257",
+      "domain#domain",
+      "asd@asd",
+      "ab..cd"};
+};
+
+/**
+ * @given some user with can_create_asset permission
+ * @when the user tries to create an asset
+ * @then asset is successfully created
+ */
+TEST_F(CreateAssetFixture, Basic) {
+  const auto asset_id = kAssetName + "#" + kDomain;
+  const auto asset_amount = "100.0";
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms({interface::permissions::Role::kCreateAsset,
+                                 interface::permissions::Role::kAddAssetQty}))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      // testing the target command
+      .sendTx(complete(baseTx().createAsset(kAssetName, kDomain, kPrecision)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      // testing that target command actually changed the state of the ledger
+      .sendTx(complete(baseTx().addAssetQuantity(asset_id, asset_amount)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
```

Comment:
Since `.done()` is called in destructor, it it required to call it here? Same for other tests.

Reply:
That was a good tradition 😄  And yes, you are right - that's time to drop all the explicit done calls.